examples/run_language_modeling.py

```diff
@@ -144,8 +146,86 @@ def __getitem__(self, i):
         return torch.tensor(self.examples[i], dtype=torch.long)
 
 
+class LazyLineByLineTextDataset(Dataset):
+    """
+    Credit: @bramvanroy for this linecache implementation.
+    """
+
+    def __init__(self, file_path):
+        self.file_path = file_path
+        self.num_entries = self._get_n_lines(self.file_path)
+
+    @staticmethod
+    def _get_n_lines(file_path):
+        with Path(file_path).resolve().open(encoding="utf-8") as fhin:
+            for line_idx, _ in enumerate(fhin, 1):
+                pass
+
+        return line_idx
+
+    def __getitem__(self, idx):
+        """
+
+        :param idx (int): the index of the line to get
+        :return (str or None): The line as a string (newline removed) or None if there is an exception.
+        """
+        # linecache starts counting from one, not zero, +1 the given index
+        line = linecache.getline(self.file_path, idx + 1).strip()
+        return line
```

Comment:
No intermediate variable assignment necessary, and probably only doing right stripping is sufficient.

```python
return linecache.getline(self.file_path, idx + 1).rstrip()
```